Installation issue: openblas error, "ld: -lto_library library filename must be 'libLTO.dylib'"

### Steps to reproduce the issue

```console
$ spack spec -I dealii
 -   dealii@9.6.2+adol-c+arborx+arpack+assimp+cgal~complex~cuda~doc~examples+examples_compile+ginkgo~gmsh+gsl+hdf5~int64~ipo+kokkos+metis+mpi+muparser~nanoflann~netcdf~oce~opencascade~optflags+p4est+petsc+platform-introspection~python+scalapack+simplex+slepc+sundials+symengine+taskflow+threads+trilinos~vtk build_system=cmake build_type=DebugRelease cxxstd=17 generator=make arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0 %fortran=gcc@15.1.0
[+]      ^adol-c@2.7.2~advanced_branching+atrig_erf~boost+doc+examples~openmp~sparse+stdczero+traceless_refcounting build_system=autotools patches:=715b878 arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
[+]          ^gnuconfig@2024-07-27 build_system=generic arch=darwin-sequoia-m1 
[e]      ^apple-clang@17.0.0 build_system=bundle arch=darwin-sequoia-m1 
 -       ^arborx@1.3~cuda~ipo+mpi~openmp~rocm+serial~sycl+trilinos build_system=cmake build_type=Release cxxstd=20 generator=make arch=darwin-sequoia-m1 %cxx=apple-clang@17.0.0
 -       ^arpack-ng@3.9.1~icb~ipo+mpi+shared build_system=cmake build_type=Release generator=make arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0 %fortran=gcc@15.1.0
 -       ^assimp@5.4.3~ipo+shared build_system=cmake build_type=Release generator=make arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
[+]          ^pkgconf@2.3.0 build_system=autotools arch=darwin-sequoia-m1 %c=apple-clang@17.0.0
 -       ^boost@1.88.0+atomic~charconv+chrono~clanglibcpp~cobalt+container~context~contract~coroutine+date_time~debug+exception~fiber+filesystem+graph~graph_parallel+icu+iostreams~json+locale+log+math~mpi~mqtt5+multithreaded~nowide~numpy~pic+program_options~python+random+regex+serialization+shared~signals2~singlethreaded~stacktrace+system~taggedlayout+test+thread+timer~type_erasure~url~versionedlayout+wave build_system=generic cxxstd=17 patches:=a440f96 visibility=hidden arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
 -           ^bzip2@1.0.8~debug~pic+shared build_system=generic arch=darwin-sequoia-m1 %c=apple-clang@17.0.0
 -           ^icu4c@76.1 build_system=autotools cxxstd=17 arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
[+]          ^xz@5.6.3~pic build_system=autotools libs:=shared,static arch=darwin-sequoia-m1 %c=apple-clang@17.0.0
[+]          ^zstd@1.5.7+programs build_system=makefile compression:=none libs:=shared,static arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
 -       ^cgal@5.6~core~demos+eigen~imageio~ipo build_system=cmake build_type=Release generator=make arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
 -           ^eigen@3.4.0~ipo~nightly~rocm build_system=cmake build_type=RelWithDebInfo generator=make arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
 -           ^gmp@6.3.0+cxx build_system=autotools libs:=shared,static arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
 -           ^mpfr@4.2.1 build_system=autotools libs:=shared,static patches:=3ec29a6 arch=darwin-sequoia-m1 %c=apple-clang@17.0.0
[+]              ^autoconf-archive@2023.02.20 build_system=autotools arch=darwin-sequoia-m1 
 -               ^texinfo@7.1 build_system=autotools arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
 -       ^cmake@3.31.8~doc+ncurses+ownlibs~qtgui build_system=generic build_type=Release arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
 -           ^curl@8.11.1~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs:=shared,static tls:=secure_transport arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
 -               ^nghttp2@1.65.0 build_system=autotools arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
[+]          ^ncurses@6.5~symlinks+termlib abi=none build_system=autotools patches:=7a351bc arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
[+]      ^compiler-wrapper@1.0 build_system=generic arch=darwin-sequoia-m1 
[e]      ^gcc@15.1.0~binutils+bootstrap~graphite~mold~nvptx~piclibs~profiled~strip build_system=autotools build_type=RelWithDebInfo languages:='c,c++,fortran' arch=darwin-sequoia-m1 
[+]      ^gcc-runtime@15.1.0 build_system=generic arch=darwin-sequoia-m1 
 -       ^ginkgo@1.9.0~cuda~develtools~full_optimizations+half_precision~hwloc~ipo~mpi~openmp~rocm~sde+shared~sycl build_system=cmake build_type=Release commit=20cfd68795f58078898da9890baa311b46845a8b generator=make arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
[+]      ^gmake@4.4.1~guile build_system=generic arch=darwin-sequoia-m1 %c=apple-clang@17.0.0
[+]      ^gsl@2.8~external-cblas+pic+shared build_system=autotools arch=darwin-sequoia-m1 %c=apple-clang@17.0.0
 -       ^hdf5@1.14.6+cxx+fortran+hl~ipo~java~map+mpi+shared~subfiling~szip~threadsafe+tools api=default build_system=cmake build_type=Release generator=make arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0 %fortran=gcc@15.1.0
 -       ^intel-tbb@2022.0.0~ipo+shared+tm build_system=cmake build_type=Release cxxstd=default generator=make arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
 -           ^hwloc@2.11.1~cairo~cuda~gl~level_zero~libudev+libxml2~nvml~opencl~pci~rocm build_system=autotools libs:=shared,static arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
[+]              ^libxml2@2.13.5~http+pic~python+shared build_system=autotools arch=darwin-sequoia-m1 %c=apple-clang@17.0.0
 -       ^metis@5.1.0~gdb~int64~ipo~no_warning~real64+shared build_system=cmake build_type=Release generator=make patches:=4991da9,93a7903 arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
 -       ^muparser@2.3.4~ipo~openmp+samples+shared~wide_char build_system=cmake build_type=Release generator=make arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
 -       ^netlib-scalapack@2.2.2~ipo~pic+shared build_system=cmake build_type=Release generator=make arch=darwin-sequoia-m1 %c=apple-clang@17.0.0 %fortran=gcc@15.1.0
 -       ^openblas@0.3.29~bignuma~consistent_fpcsr+dynamic_dispatch+fortran~ilp64+locking+pic+shared build_system=makefile patches:=1a2401f symbol_suffix=none threads=none arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0 %fortran=gcc@15.1.0
 -       ^openmpi@5.0.8+atomics~cuda~debug+fortran~gpfs~internal-hwloc~internal-libevent~internal-pmix~ipv6~java~lustre~memchecker~openshmem~rocm~romio+rsh~static~two_level_namespace+vt+wrapper-rpath build_system=autotools fabrics:=none romio-filesystem:=none schedulers:=none arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0 %fortran=gcc@15.1.0
 -           ^autoconf@2.72 build_system=autotools arch=darwin-sequoia-m1 
 -           ^automake@1.16.5 build_system=autotools arch=darwin-sequoia-m1 %c=apple-clang@17.0.0
 -           ^libevent@2.1.12+openssl build_system=autotools arch=darwin-sequoia-m1 %c=apple-clang@17.0.0
 -               ^openssl@3.4.1~docs+shared build_system=generic certs=mozilla arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
[+]                  ^ca-certificates-mozilla@2025-05-20 build_system=generic arch=darwin-sequoia-m1 
 -           ^libtool@2.4.7 build_system=autotools arch=darwin-sequoia-m1 %c=apple-clang@17.0.0
 -               ^findutils@4.10.0 build_system=autotools patches:=440b954 arch=darwin-sequoia-m1 %c=apple-clang@17.0.0
 -           ^openssh@9.9p1+gssapi build_system=autotools patches:=3f06fc0,d886b98 arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
 -               ^krb5@1.21.3+shared build_system=autotools arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
 -               ^libedit@3.1-20240808 build_system=autotools arch=darwin-sequoia-m1 %c=apple-clang@17.0.0
 -               ^libxcrypt@4.4.38~obsolete_api build_system=autotools arch=darwin-sequoia-m1 %c=apple-clang@17.0.0
 -           ^perl@5.40.0+cpanm+opcode+open+shared+threads build_system=generic arch=darwin-sequoia-m1 %c=apple-clang@17.0.0
[+]              ^berkeley-db@18.1.40+cxx~docs+stl build_system=autotools patches:=26090f4,b231fcc arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
 -               ^gdbm@1.23 build_system=autotools arch=darwin-sequoia-m1 %c=apple-clang@17.0.0
 -           ^pmix@5.0.5~munge~python build_system=autotools arch=darwin-sequoia-m1 %c=apple-clang@17.0.0
 -       ^p4est@2.8.7+mpi~openmp build_system=autotools arch=darwin-sequoia-m1 %c=apple-clang@17.0.0
 -       ^petsc@3.23.4~X~batch~cgns~complex~cuda~debug+double+examples~exodusii~fftw+fortran~giflib+hdf5~hpddm~hwloc+hypre~int64~jpeg~knl~kokkos~libpng~libyaml~memkind+metis~mkl-pardiso~mmg~moab~mpfr+mpi+mumps~openmp+p4est~parmmg+ptscotch~random123~rocm~saws+scalapack+shared~strumpack+suite-sparse+superlu-dist~sycl~tetgen~trilinos~valgrind~zoltan build_system=generic clanguage=C memalign=none arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0 %fortran=gcc@15.1.0
[+]          ^diffutils@3.10 build_system=autotools arch=darwin-sequoia-m1 %c=apple-clang@17.0.0
[+]              ^libiconv@1.18 build_system=autotools libs:=shared,static arch=darwin-sequoia-m1 %c=apple-clang@17.0.0
 -           ^hypre@2.33.0~caliper~complex~cuda~debug+fortran~gptune~gpu-aware-mpi~gpu-profiling~int64~internal-superlu+lapack~magma~mixedint+mpi~openmp~rocm+shared~superlu-dist~sycl~umpire~unified-memory build_system=autotools precision=double arch=darwin-sequoia-m1 %c=apple-clang@17.0.0 %fortran=gcc@15.1.0
 -           ^mumps@5.8.0~blr_mt+complex+double+float~incfort~int64+metis+mpi~openmp+parmetis~ptscotch~scotch+shared build_system=generic patches:=14f0694 arch=darwin-sequoia-m1 %c=apple-clang@17.0.0 %fortran=gcc@15.1.0
 -           ^parmetis@4.0.3~gdb~int64~ipo+shared build_system=cmake build_type=Release generator=make patches:=4f89253,50ed208,704b84f arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
 -           ^python@3.13.5+bz2+ctypes+dbm~debug+libxml2+lzma~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
[e]              ^apple-libuuid@1353.100.2 build_system=bundle arch=darwin-sequoia-m1 
[+]              ^expat@2.7.1~libbsd build_system=autotools arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
 -               ^gettext@0.23.1+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
 -                   ^tar@1.35 build_system=autotools zip=pigz arch=darwin-sequoia-m1 %c=apple-clang@17.0.0
[+]                      ^pigz@2.8 build_system=makefile arch=darwin-sequoia-m1 %c=apple-clang@17.0.0
[+]              ^libffi@3.4.8 build_system=autotools arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
 -               ^readline@8.2 build_system=autotools patches:=1ea4349,24f587b,3d9885e,5911a5b,622ba38,6c8adf8,758e2ec,79572ee,a177edc,bbf97f1,c7b45ff,e0013d9,e065038 arch=darwin-sequoia-m1 %c=apple-clang@17.0.0
 -               ^sqlite@3.46.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=darwin-sequoia-m1 %c=apple-clang@17.0.0
 -           ^scotch@7.0.7+compression+esmumps~int64~ipo~metis+mpi~mpi_thread~noarch+shared+threads build_system=cmake build_type=Release generator=make arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0 %fortran=gcc@15.1.0
 -               ^bison@3.8.2~color build_system=autotools arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
 -               ^flex@2.6.4+lex~nls build_system=autotools patches:=f8b85a0 arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
 -                   ^help2man@1.49.3 build_system=autotools arch=darwin-sequoia-m1 %c=apple-clang@17.0.0
 -           ^superlu-dist@9.1.0~cuda~int64~ipo~openmp+parmetis~rocm+shared build_system=cmake build_type=Release generator=make arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0 %fortran=gcc@15.1.0
 -       ^slepc@3.23.2+arpack~blopex~cuda~hpddm~rocm build_system=generic arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0 %fortran=gcc@15.1.0
 -       ^suite-sparse@7.3.1~cuda~graphblas~openmp+pic build_system=generic arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0 %fortran=gcc@15.1.0
 -           ^m4@1.4.20+sigsegv build_system=autotools arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
[+]              ^libsigsegv@2.14 build_system=autotools arch=darwin-sequoia-m1 %c=apple-clang@17.0.0
 -       ^sundials@6.7.0+ARKODE+CVODE+CVODES+IDA+IDAS+KINSOL~cuda+examples+examples-install~f2003~fcmix+generic-math~ginkgo~hypre~int64~ipo~klu~kokkos~kokkos-kernels~lapack~magma~monitoring+mpi~openmp~petsc~profiling~pthread~raja~rocm+shared+static~superlu-dist~superlu-mt~sycl~trilinos build_system=cmake build_type=Release cstd=99 cxxstd=14 generator=make logging-level=2 logging-mpi=OFF precision=double arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0 %fortran=gcc@15.1.0
 -       ^symengine@0.9.0~boostmp~flint~ipo~llvm+mpc+mpfr~openmp~piranha+shared+thread_safe build_system=cmake build_type=Release generator=make arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
 -           ^cereal@1.3.2~ipo build_system=cmake build_type=Release generator=make patches:=2dfa0bf arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
 -           ^mpc@1.3.1 build_system=autotools libs:=shared,static arch=darwin-sequoia-m1 %c=apple-clang@17.0.0
 -       ^taskflow@3.7.0~ipo build_system=cmake build_type=Release generator=make arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0 %fortran=gcc@15.1.0
 -       ^trilinos@13.4.1~adelus~adios2+amesos+amesos2+anasazi+aztec~basker+belos~boost~chaco~complex~cuda~cuda_constexpr~cuda_rdc~debug~dtk+epetra+epetraext~epetraextbtf~epetraextexperimental~epetraextgraphreorderings~exodus+explicit_template_instantiation~float+fortran~gtest~hdf5~hypre+ifpack+ifpack2~intrepid~intrepid2~ipo~isorropia+kokkos+mesquite~minitensor+ml+mpi+muelu+mumps~nox~openmp~pamgen~panzer~phalanx~piro~python~rocm~rocm_rdc+rol~rythmos+sacado~scorec~shards+shared~shylu~stk~stokhos~stratimikos~strumpack+suite-sparse~superlu+superlu-dist~teko~tempus~test~thyra+tpetra~trilinoscouplings~wrapper~x11~zoltan~zoltan2 build_system=cmake build_type=Release cxxstd=14 generator=make gotype=long_long patches:=016a979 arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0 %fortran=gcc@15.1.0
[+]      ^zlib-ng@2.2.4+compat+new_strategies+opt+pic+shared build_system=autotools arch=darwin-sequoia-m1 %c,cxx=apple-clang@17.0.0
```


### Error message

<details><summary>Error message</summary>

<pre>
5 errors found in build log:
     7340    zblat3.f:1315:44:
     7341
     7342     1315 |       COMPLEX*16         ALPHA, ALS, BETA, BETS
     7343          |                                            ^~~~
     7344    note: 'IMAGPART_EXPR <bets>' was declared here
     7345    ld: -lto_library library filename must be 'libLTO.dylib'
  >> 7346    collect2: error: ld returned 1 exit status
  >> 7347    make[1]: *** [Makefile:320: sblat1] Error 1
     7348    make[1]: *** Waiting for unfinished jobs....
     7349    ld: -lto_library library filename must be 'libLTO.dylib'
  >> 7350    collect2: error: ld returned 1 exit status
  >> 7351    make[1]: *** [Makefile:331: dblat1] Error 1
  >> 7352    make: *** [Makefile:171: tests] Error 2
</pre></details>


### Information on your system

* **Spack:** 1.0.0.dev0 (https://github.com/spack/spack/commit/c64f33dc6a9263757623467684b1326967c47e96)
* **Builtin repo:** https://github.com/spack/spack-packages/commit/98bd916bc8a3c97c62fd1feab0aa4c6754b86419
* **Python:** 3.9.6
* **Platform:** darwin-sequoia-m1

```
$ cat ~/.spack/packages.yaml 
packages:
  gcc:
    externals:
    - spec: gcc@15.1.0 languages:='c,c++,fortran'
      prefix: /opt/homebrew
      extra_attributes:
        compilers:
          c: /opt/homebrew/bin/gcc-15
          cxx: /opt/homebrew/bin/g++-15
          fortran: /opt/homebrew/bin/gfortran
  apple-clang:
    externals:
    - spec: apple-clang@17.0.0
      prefix: /usr
      extra_attributes:
        compilers:
          c: /usr/bin/clang
          cxx: /usr/bin/clang++
# python:
#   externals:
#     - spec: python@3.12.7
#       prefix: /opt/homebrew
#       buildable: false
  dealii:
    require: ~examples +simplex ~nanoflann ~gmsh ~python ~vtk ~opencascade
  hdf5:
    require: +fortran +cxx +hl
  petsc:
    require: +mumps +scalapack +suite-sparse +superlu-dist +p4est +ptscotch
  cgns:
    require: +fortran
  trilinos:
    require: +rol +mesquite +mumps +suite-sparse +superlu-dist
  libmesh:
    require: +boost +eigen +fparser +hdf5 +laspack +metis +perflog +petsc +slepc +tetgen +triangle
  vtk:
    require: +python
  all:
    providers:
      mpi: [openmpi]
      blas: [openblas]
      lapack: [openblas]
```

### Additional information

[spack-build-out.txt](https://github.com/user-attachments/files/21261870/spack-build-out.txt)

[spack-build-env.txt](https://github.com/user-attachments/files/21261875/spack-build-env.txt)

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

When I tried again, it is now installing v0.3.30 but I get the same error

```
     12685    note: 'IMAGPART_EXPR <bets>' was declared here
     12686    ld: -lto_library library filename must be 'libLTO.dylib'
  >> 12687    collect2: error: ld returned 1 exit status
  >> 12688    make[1]: *** [Makefile:320: sblat1] Error 1
     12689    make[1]: *** Waiting for unfinished jobs....
     12690    ld: -lto_library library filename must be 'libLTO.dylib'
  >> 12691    collect2: error: ld returned 1 exit status
  >> 12692    make[1]: *** [Makefile:331: dblat1] Error 1
     12693    ld: -lto_library library filename must be 'libLTO.dylib'
  >> 12694    collect2: error: ld returned 1 exit status
  >> 12695    make[1]: *** [Makefile:349: cblat1] Error 1
  >> 12696    make: *** [Makefile:176: tests] Error 2
```

What am I doing wrong ?

@Chrismarsh well nuts. I now see this:
```
     12522
     12523     1315 |       COMPLEX*16         ALPHA, ALS, BETA, BETS
     12524          |                                            ^~~~
     12525    note: 'IMAGPART_EXPR <bets>' was declared here
     12526    ld: -lto_library library filename must be 'libLTO.dylib'
  >> 12527    collect2: error: ld returned 1 exit status
  >> 12528    make[1]: *** [Makefile:320: sblat1] Error 1
     12529    make[1]: *** Waiting for unfinished jobs....
     12530    ld: -lto_library library filename must be 'libLTO.dylib'
  >> 12531    collect2: error: ld returned 1 exit status
  >> 12532    make[1]: *** [Makefile:331: dblat1] Error 1
  >> 12533    make: *** [Makefile:176: tests] Error 2
```

it's a bit mangled, but it's the same thing.

I just tried installing OpenBLAS 0.3.30 with spack using GCC 15.2 and I get that error.

And I do have the latest xcode installed:
```
❯ xcodebuild -version
Xcode 26.0.1
Build version 17A400
```

--- a/repos/spack_repo/builtin/packages/openblas/openblas-0.3.30-apple-LTO.patch
+++ b/repos/spack_repo/builtin/packages/openblas/openblas-0.3.30-apple-LTO.patch
@@ -1,0 +1,13 @@
+diff --git a/Makefile.system b/Makefile.system
+index 38646c3c6..e81ccf083 100644
+--- a/Makefile.system
++++ b/Makefile.system
+@@ -436,7 +436,7 @@ ifeq (x$(XCVER), x 15)
+ CCOMMON_OPT += -Wl,-ld_classic
+ FCOMMON_OPT += -Wl,-ld_classic
+ endif
+-ifeq (x$(XCVER), x 16)
++ifeq ($(shell [ $(XCVER) -ge 16 ] && echo yes),yes)
+ ifeq ($(F_COMPILER), GFORTRAN)
+ override CEXTRALIB := $(filter-out(-lto_library, $(CEXTRALIB))) 
+ endif

--- a/repos/spack_repo/builtin/packages/openblas/package.py
+++ b/repos/spack_repo/builtin/packages/openblas/package.py
@@ -162,6 +162,9 @@ class Openblas(CMakePackage, MakefilePackage):
     # https://github.com/OpenMathLib/OpenBLAS/pull/1703
     patch("openblas-0.3.2-cmake.patch", when="@0.3.1:0.3.2")
 
+    # https://github.com/OpenMathLib/OpenBLAS/issues/5473
+    patch("openblas-0.3.30-apple-LTO.patch", when="@0.3.30 platform=darwin")
+
     # Disable experimental TLS code that lead to many threading issues
     # https://github.com/OpenMathLib/OpenBLAS/issues/1735#issuecomment-422954465
     # https://github.com/OpenMathLib/OpenBLAS/issues/1761#issuecomment-421039174

